src/manage_nsfs/manage_nsfs_validations.js

```diff
@@ -447,6 +448,23 @@ async function validate_bucket_args(config_fs, data, action) {
     }
 }
 
+/**
+ * When setting notifications, we are supposed to send a test notification.
+ * If this test fails, we fail the user's request.
+ * @param {object} config_fs
+ * @param {object} notifications bucket's notification conf
+ * @returns Error encountered during notification test, if any
+ */
+async function validate_bucket_notifications(config_fs, notifications) {
+    if (!notifications) {
+        return;
+    }
+    const test_notif_err = await notifications_util.test_notifications(notifications, config_fs.config_root);
```

Comment:
If an error is thrown how come we didn't wrap it with try-catch clause?
Maybe worth adding a comment here so it will be more clear and also a JSDoc on the function `test_notifications`.

Reply:
Error is not thrown, it is returned.
I will add a comment.